### PR TITLE
[5.4] add schedule:list command

### PR DESCRIPTION
**Pull request by jaumesala**
As discused on #20555 this command shows basic information of the scheduled tasks. The expresion string is written "as is" in the event class, so the minutes are not correlated with the cron expression (as @laurencei pointed out). Still I think this command can be helpful to developers looking for a fast way to see what's being sent to the task scheduler. 

schedule-list

### CODE REVIEW

**Comment by decadence** [comment]
Related: https://github.com/laravel/framework/pull/19811

File: src/Illuminate/Console/Scheduling/ScheduleListCommand.php [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+     * Execute the console command.
+     *
+     * @return void
+     */
+    public function fire()
+    {
+        $events = $this->getEvents();
+
+        if (empty($events)) {
+            return $this->info('Your application has no scheduled commands to run.');
```
Comment (laurencei) [inline src/Illuminate/Console/Scheduling/ScheduleListCommand.php]: Maybe different message? "There are no scheduled commands to list" or something.

The error message sounds like it's trying to run them (which it is not)

**Comment by taylorotwell** [comment]
I still feel like the simplest way to view your schedule is to simply look at your schedule calls in the Console kernel itself.

**Comment by jaumesala** [comment]
I don't know, I've been recently working on a project that kept me all the time opening the Console Kernel file, a command like this would have been very useful. Looks like @decadence was also trying to achieve the same results (I didn't see your pull request sorry).
What if we remove the *expression* column and focus on listing what's in the Console Kernel?
We can change the *expression* for *next execution* like @decadence did in his pull request.

File: src/Illuminate/Console/Scheduling/ScheduleListCommand.php (hunk 1 above)
Comment (jaumesala) [inline src/Illuminate/Console/Scheduling/ScheduleListCommand.php]: Yes, You are right.

**Comment by decadence** [comment]
I still think this command will be useful for quick view on schedule via console.
@jaumesala your PR more informative and we could add my "Next Execution date" column to it.
Unfortunately @taylorotwell has another opinion.

**Comment by jaumesala** [comment]
Even if this PR is closed, I've just updated my branch with the discussed points. I have added @decadence idea and corrected the empty message as suggested by @laurencei. 

schedule-list2

**Comment by decadence** [comment]
@jaumesala thanks. I will save it for the future use ;)

**Comment by RikSomers** [comment]
@taylorotwell To be honest, the kernel file can sometimes get rather big and complicated, so having a command to show which tasks are scheduled can be extremely useful in my opinion.

**Comment by laurencei** [comment]
This is a perfect snipped of code that can be easily wrapped in a package.

In fact - I quickly googled and one already exists: https://github.com/hmazter/laravel-schedule-list

So if you want the functionality - your best bet is just install the package...

**Comment by thecrypticace** [comment]
Heh, I was just looking for something like this yesterday to verify that something was indeed running at the time I thought it was supposed to. Thought Laravel included it by default but at least there's a package for it. 👍

**Pull request closed without merging**